### PR TITLE
Preload models in parallel

### PR DESCRIPTION
**Pull request by kkaneda**

### CODE REVIEW

File: engine/cmd/run.go [hunk 1]
```diff
@@ -138,12 +138,16 @@ func run(ctx context.Context, c *config.Config) error {
...
-				return err
-			}
+			go func() {
+				log.Printf("Preloading model %q", id)
+				if err := syncer.PullModel(ctx, id); err != nil {
+					errCh <- err
+					return
+				}
+				log.Printf("Completed the preloading of %q", id)
+				return
```
Comment (kkaneda) [inline engine/cmd/run.go:148]:
```suggestion
```

**Review by Ladicle — APPROVED** [review]
It would be nice to eventually limit the number of parallels.

**Comment by kkaneda** [comment]
Agreed. Let me merge this after more testing.

**Comment by kkaneda** [comment]
I'll also add the limit to 2 for now.

**Pull request merged**